airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from time import sleep
+
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+from airflow.exceptions import AirflowException
+from airflow.contrib.hooks.kubernetes_hook import KubernetesHook
+
+
+class KubernetesPodOperator(BaseOperator):
+    """
+    Deploys docker container to k8s pod and waits for its completion
+
+    :param name: Name of the pod
+    :type name: string
+    :param namespace: Namespace the pod will be deployed to
+    :type namespace: string
+    :param image: Fully qualified name of the image in form
+        of repo/image:tag
+    :type image: string
+    :param command: Commands to execute in the image,
+        default image command will be executed if none supplied
+    :type command: string or list
+    :param op_args: Arguments for the command
+    :type op_args: list
+    :param wait: Wait for the completion. Default True. If set to false
+        the operator waits for the pod to start running to ensure
+        successful creation.
+    :type wait: boolean
+    :param cleanup: Perform cleanup on completion.
+        Allowed values: Always, Never, OnSuccess, OnFailure.
+        Default Always. Settign wait == False forces Never,
+        as cleanup can be only performed on terminated container.
+    :param labels: Labels and presets to apply to the pod.
+    :type labels: dict
+    :param conn_id: Id of pre-defined k8s connection. Currently not used,
+        as only preconfigured environment with kube config or env variables
+        is supported.
+    :type conn_id: string
+    :param poke_interval: Interval between checking the status in seconds
+    :type poke_interval: integer
+    """
+    template_fields = ('name', 'command', 'op_args', "namespace")
+    ui_color = '#f0ede4'
+
+    @apply_defaults
+    def __init__(
+            self,
+            name,
+            namespace,
+            image,
+            command=None,
+            op_args=None,
+            wait=True,
+            cleanup="Always",
+            labels=None,
+            conn_id="k8s_default",
+            poke_interval=3,
+            *args, **kwargs):
+        super(KubernetesPodOperator, self).__init__(*args, **kwargs)
+        self.image = image
+        self.name = name
```

Comment:
pod names must be unique within the namespace. If a pod failed and wasn't cleaned up, this might block execution. Not sure if it should be the executors task of guarannteeing uniqueness or if the user should take care of this in the DAG.